dagger/stores/memorystore.nim

```diff
@@ -7,54 +7,72 @@
 ## This file may not be copied, modified, or distributed except according to
 ## those terms.
 
+{.push raises: [Defect].}
+
 import std/sequtils
+
 import pkg/chronos
 import pkg/libp2p
+import pkg/chronicles
+import pkg/questionable
 
-import ../stores/blockstore
+import ./blockstore
 import ../blocktype
 
 export blockstore
 
+logScope:
+  topics = "dagger memstore"
+
 type
   MemoryStore* = ref object of BlockStore
-    blocks: seq[Block] # TODO: Should be an LRU cache
+    blocks: seq[?Block] # TODO: Should be an LRU cache
 
-method getBlocks*(
-  s: MemoryStore,
-  cids: seq[Cid]): Future[seq[Block]] {.async.} =
+method getBlock*(
+  b: MemoryStore,
+  cid: Cid): Future[?Block] {.async.} =
   ## Get a block from the stores
   ##
 
-  var res: seq[Block]
-  for c in cids:
-    res.add(s.blocks.filterIt( it.cid == c ))
+  trace "Getting block", cid
+  let found = b.blocks.filterIt(
+    (!it).cid == cid
```

Comment:
I'd avoid using `!` in implementation code (it's fine in test code). Perhaps replace it with:
```nim
it.?cid == cid.some
```
Same holds for line 53 and 72.

Reply:
I'm a bit concerned with all the wrapping and unwrapping that goes on here, it definitely feels a bit weird, but for safety you're probably right that it's better style to use `?`.